File: 백준/Silver/10828. 스택/스택.py
import sys
N = int(sys.stdin.readline().rstrip())
stack = []

for _ in range(N):
    command = sys.stdin.readline().split()

    if command[0] == 'push': # push 명령어인 경우 command[1]도 있음
        stack.append(command[1]) # 가장 위에 정수를 추가
    elif command[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif command[0] == 'size':
        print(len(stack))
    elif command[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif command[0] == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: 백준/Silver/10828. 스택/스택.java
import java.util.*;

public class Main {
    public static int[] stack; // 스택 배열
    public static int size = 0; // 현재 스택의 크기

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();

        int N = sc.nextInt(); // 명령어 반복 횟수
        stack = new int[N]; // 스택 배열 크기 설정

        for (int i = 0; i < N; i++) {
            String command = sc.next();

            switch (command) {
                case "push":
                    push(sc.nextInt());
                    break;

                case "pop":
                    sb.append(pop()).append('\n');
                    break;

                case "size":
                    sb.append(size()).append('\n');
                    break;

                case "empty":
                    sb.append(empty()).append('\n');
                    break;

                case "top":
                    sb.append(top()).append('\n');
                    break;
            }
        }
        System.out.print(sb);
    }

    private static void push(int num) {
        stack[size] = num;
        size++;
    }

    private static int pop() {
        if (size == 0) {
            return -1;
        } else {
            size--;
            return stack[size];
        }
    }

    private static int size() {
        return size;
    }

    private static int empty() {
        return size == 0 ? 1 : 0;
    }

    private static int top() {
        if (size == 0) {
            return -1;
        } else {
            return stack[size - 1];
        }
    }
}
